backend/routes/user.js

```diff
@@ -0,0 +1,113 @@
+import express from 'express';
+import bodyParser from 'body-parser';
+import connection from '../database';
+import config from '../config/config';
+
+const router = express.Router();
+router.use(bodyParser.urlencoded({ extended: true }));
+
+// Add new user to database. Used when adding a new reservation, so that a user can identify themself.
+router.post('/', async (req, res) => {
+  const { body } = req;
+
+  if (!body.firstName || !body.lastName || !body.phone || !body.email) {
+    res.status(400).json({ error: '/user POST endpoint needs firstName, lastName, phone, and email body params' });
+    return;
+  }
+
+  const { error, result } = await connection.asyncQuery(
+    'INSERT INTO USER(FirstName, LastName, Phone, Email) VALUES (?, ?, ?, ?);',
+    [body.firstName, body.lastName, body.phone, body.email]
+  );
+
+  if (error) {
+    res.status(400).json({ error });
+    return;
+  }
+
+  const userID = result.insertId;
+  res.json({ result: 'Added user', userID });
+});
+
+// Update user in database.
+router.put('/', async (req, res) => {
+  const { body } = req;
+
+  if (!body.reservationID || (!body.firstName && !body.lastName && !body.phone && !body.email)) {
+    res
+      .status(400)
+      .json({ error: '/user PUT endpoint needs firstName, lastName, phone, OR email, AND reservationID body params' });
+    return;
+  }
+
+  const { error, result } = await connection.asyncQuery(
+    'SELECT * FROM USER JOIN RESERVATION ON USER.ID = RESERVATION.UserID WHERE RESERVATION.ID = ?',
+    [body.reservationID]
+  );
+
+  if (error) {
+    res.status(400).json({ error });
+    return;
+  }
+
+  let newFirstName;
+  let newLastName;
+  let newPhone;
+  let newEmail;
+
+  // For each parameter, check if it has been provided in endpoint, otherwise use previous value.
+  // The mock SQL doesn't return a proper result, so a default value is assigned when testing
+  if (body.firstName) {
+    newFirstName = body.firstName;
```

Comment:
Is it possible to refactor the if statements?

Reply:
How would you suggest doing that? I kept them all separate so that the endpoint can take any number of the fields and set all the other fields to the previous value. I haven’t used js before so if there’s a better way of doing that let me know

Reply:
I didn't end up using a for loop, but I did refactor them a different way that Jacob showed me. Take a look when you can :)